crates/core/tests/helpers/deploy_helpers.rs

```diff
@@ -0,0 +1,476 @@
+use std::collections::HashMap;
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+use std::sync::Arc;
+
+use bytes::BytesMut;
+use dojo_test_utils::sequencer::TestSequencer;
+use dotenv::dotenv;
+use ethers::abi::{Abi, Tokenize};
+use kakarot_rpc_core::client::constants::CHAIN_ID;
+use reth_primitives::{
+    sign_message, Address, Bytes, Transaction, TransactionKind, TransactionSigned, TxEip1559, H256, U256,
+};
+use serde::Deserialize;
+use starknet::accounts::{Account, Call, ConnectedAccount, SingleOwnerAccount};
+use starknet::contract::ContractFactory;
+use starknet::core::chain_id;
+use starknet::core::types::contract::legacy::LegacyContractClass;
+use starknet::core::types::{
+    BlockId, BlockTag, FieldElement, FunctionCall, InvokeTransactionReceipt, MaybePendingTransactionReceipt,
+    TransactionReceipt,
+};
+use starknet::core::utils::{get_contract_address, get_selector_from_name};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+use starknet::signers::{LocalWallet, SigningKey};
+use toml;
+use url::Url;
+
+use super::constants::FEE_TOKEN_ADDRESS;
+
+macro_rules! root_project_path {
+    ($relative_path:expr) => {{
+        let crate_root = std::path::Path::new(env!("CARGO_MANIFEST_DIR"));
+        // this is assuming this code will always be ran in the context where the CARGO_MANIFEST_DIR
+        // <system>/kakarot-rpc/crates/core
+        // this is done so we can have more flexibility in filepaths the artifacts are located
+        // so multiple crates can access them, if necessary
+        let project_root = crate_root.parent().unwrap().parent().unwrap();
+        let full_path = project_root.join($relative_path);
+        full_path
+    }};
+}
+
+#[derive(Debug, Deserialize)]
+struct Profile {
+    default: DefaultProfile,
+}
+
+#[derive(Debug, Deserialize)]
+struct DefaultProfile {
+    out: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct FoundryConfig {
+    profile: Profile,
+}
+
+fn get_foundry_default_out(file_path: &std::path::Path) -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
could you add some context, using rustdoc comments?

also, I think you could technically hardcode the foundry.toml filename / path, since we're forcing this util to be coupled with foundry types `FoundryConfig`